frontend/src/store/wallet.ts

```diff
@@ -129,6 +132,9 @@ export default function useWalletStore() {
             window.location.reload();
           });
           wallet.provider.on('chainChanged', () => {
+            if (wallet.label === 'WalletConnect' && !lastWallet.value) {
+              setLastWallet('WalletConnect');
+            }
```

Comment:
I added this because when I was testing the chainChanged event was being called before the lastWallet was saved. And if there was no lastWallet value than the app to look as a if a user was not logged in. Open to other solutions as well

Reply:
> And if there was no lastWallet value than the app to look as a if a user was not logged in

Sorry, not sure I understand what you mean here?

Also, based on that explanation, then why do we need the `wallet.label === 'WalletConnect'` check? It sounds like we'd only need the `!lastWallet.value` check

Reply:
Oh sorry maybe a visual would be more clear.

[Screencast from 2023-07-05 21-02-34.webm](https://github.com/ScopeLift/umbra-protocol/assets/12705146/a5977d14-d3d2-4a35-981b-55f0a4272c37)

The reason I added the label check was so this behavior would only happen for WalletConnect as it doesn't seem to be an issue when connecting with metamask.